crates/context/src/lib.rs

```diff
@@ -1421,4 +1421,15 @@ impl ContextManager {
             .map_err(|err| eyre::eyre!("Failed to fetch context storage entries: {}", err))?;
         Ok(response)
     }
+
+    pub async fn get_proxy_id(&self, context_id: ContextId) -> EyreResult<String> {
+        let handle = self.store.handle();
+        let Some(context_config) = handle.get(&ContextConfigKey::new(context_id))? else {
+            bail!("Context not found");
+        };
+
+        let proxy_contract = context_config.proxy_contract.as_ref().into();
```

Comment:
here you want to convert `Box<str>` -> `String`

`String` [impls](https://doc.rust-lang.org/std/string/struct.String.html#impl-From%3CBox%3Cstr%3E%3E-for-String) `From<Box<str>>`


```suggestion
        let proxy_contract = context_config.proxy_contract.into();
```

the `.as_ref` would've made it `Box<str>` -> `&str` -> `String`, of which the `&str` -> `String` part would've involved an allocation